go/vt/topo/keyspace.go

```diff
@@ -270,26 +273,77 @@ func (ts *Server) UpdateKeyspace(ctx context.Context, ki *KeyspaceInfo) error {
 	return nil
 }
 
-// FindAllShardsInKeyspace reads and returns all the existing shards in
-// a keyspace. It doesn't take any lock.
-func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string) (map[string]*ShardInfo, error) {
+// FindAllShardsInKeyspaceConfig controls the behavior of
+// Server.FindAllShardsInKeyspace.
+type FindAllShardsInKeyspaceConfig struct {
+	// Concurrency controls the maximum number of concurrent calls to GetShard.
+	// If unspecified, Concurrency is set to 1.
+	Concurrency int
+}
+
+// FindAllShardsInKeyspace reads and returns all the existing shards in a
+// keyspace. It doesn't take any lock.
+//
+// If cfg is non-nil, it is used to configure the method's behavior. Otherwise,
+// a default configuration is used.
+func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string, cfg *FindAllShardsInKeyspaceConfig) (map[string]*ShardInfo, error) {
+	// Apply any necessary defaults.
+	if cfg == nil {
+		cfg = &FindAllShardsInKeyspaceConfig{}
+	}
+	if cfg.Concurrency == 0 {
+		cfg.Concurrency = 1
+	}
```

Comment:
extreme nit: can technically skip one extra statement by setting the default when `cfg` is `nil`

```suggestion
	if cfg == nil {
		cfg = &FindAllShardsInKeyspaceConfig{Concurrency: 1}
	}
	if cfg.Concurrency == 0 {
		cfg.Concurrency = 1
	}
```

Reply:
Ack, the one reason I didn't do this is because you'd have to duplicate any future options in the same manner. I'll leave this as is.